packages/remix-dev/compiler.ts

```diff
@@ -355,6 +366,11 @@ async function createBrowserBuild(
     NodeModulesPolyfillPlugin(),
   ];
 
+  let yarnPnpPlugin = await getYarnPnpPlugin();
+  if (yarnPnpPlugin) {
+    plugins.push(yarnPnpPlugin);
+  }
```

Comment:
I like this one, although TypeScript then complains when passing this array to `esbuild.build`, because it now is of type `(esbuild.Plugin | null)[]`.
Writing `let plugins = [/* ... */].filter((p): p is esbuild.Plugin => !!p);` works, although it's a bit more verbose.

Reply:
maybe if you add `?? undefined` after the nullable item it’ll satisfy the type checker